docs/client-side/yeoman.rst

```diff
@@ -0,0 +1,188 @@
+- Console Application: Another minimal application that outputs "hello world" to the console.
+- Web Application: A complete MVC web application with a simple home page and examples for managing accounts and login.  
+- Web API Application: An ASP.NET 5 web service with one controller.   
+- Nancy ASP.NET Application: A light-weight HTTP service with one module.
+- Class Library: Creates a DLL that can be referenced by other ASP.NET projects.
+
+This walk-through demonstrates how to use Yeoman to generate an ASP.NET 5 web application. 
+ 
+1.  Follow the instructions on http://yeoman.io/learning/ to install **yo** and other required tools. 
+
+2.	On the command line, install the ASP.NET generator:  
+
+.. code-block:: console
+
+ npm install -g generator-aspnet
+
+.. note:: The **–g** flag installs the generator globally so that you can use it from any path on your system.
+ 
+3.	Make a new directory where your project will be generated:
+
+.. code-block:: console
+
+ mkdir c:\MyYo
+
+4.	On the command line, make the new directory the current directory.
+
+.. code-block:: console
+ 
+ cd c:\MyYo
+
+5.	Run the ``yo`` command and pass the name of the generator.
+
+.. code-block:: console 
+
+ yo aspnet
+
+6.	The aspnet generator displays a menu. Select the **Web Application** and press ``Enter``.
+
+.. image:: yeoman/_static/yeoman-yo-aspnet.png
+
+7.	Provide an application name "MyWebApp" and press ``Enter``.
+
+.. image:: yeoman/_static/yeoman-yo-aspnet-appname.png
+
+Yeoman will create the project and supporting files. 
+
+.. image:: yeoman/_static/yeoman-yo-aspnet-created.png
+
+Client-Side Build Support
+-------------------------
+
+The `ASPNET generator <https://www.npmjs.com/package/generator-aspnet>`_ creates support files to configure client-side build tools. `Grunt or Gulp task runners <http://docs.asp.net/en/latest/client-side/grunt-gulp.html>`_ files are added to your project to automate build tasks for Web Application projects. By default, aspnet-generator creates **gruntfile.js** to run tasks. Running the generator with the **--gulp** argument generates **gulpfile.js** instead.
+
+.. code-block:: console 
+
+ yo aspnet --gulp
+ 
+The generator also configures **package.json** to load Grunt or Gulp and configures **project.json** to restore client-side packages
+using the `Bower client-side package manager <http://docs.asp.net/en/latest/client-side/bower.html>`_.  
+
+Building and Running from Visual Studio
+---------------------------------------
+
+You can load your generated ASP.NET 5 web project directly into Visual Studio 2015, then build and run your project from there.
+
+1. Open Visual Studio 2015. From the File menu select :menuselection:`Open --> Project/Solution`.
+
+2. In the Open Project dialog, navigate to the ``project.json`` file, select it and click the **Open** button. In the Solution Explorer, the project should look something like the screenshot below.
+
+ .. image:: yeoman/_static/yeoman-solution.png
```

Comment:
this screen shot shows the package restore in progress (i.e. Restoring . . .). Can we instead have an image with package restore already completed?